src/context/auth.js

```diff
@@ -23,39 +23,77 @@ const AuthProvider = ({ children }) => {
         }
     }, [location.state?.from?.pathname, navigate])
 
-	const handleLogin = async (email, password) => {
-		const res = await login(email, password)
-
+	const handleLogin = async (email, password, setErrors) => {
+
+		const passwordRequirments = 'Password must be at least 8 characters long and include at least one uppercase letter, one number, and one special character.'
```

Comment:
Let's put these errors in a `utils/errors.js` file so they're not scattered around the app